schema-registry/rest-service/src/main/java/com/hortonworks/registries/schemaregistry/webservice/SchemaRegistryResource.java

```diff
@@ -983,4 +1019,79 @@ public Response deleteSchemaVersion(@ApiParam(value = "Schema name", required =
+    @Path("/branch/version/{version}/create")
+    @ApiOperation(value = "Fork a new schema branch given its schema name and version id", tags = OPERATION_GROUP_SCHEMA)
+    @UnitOfWork
+    public Response createSchemaBranch( @ApiParam(value = "Details about schema version",required = true) @PathParam("version") Long schemaVersionId,
+                                        @ApiParam(value = "Schema Branch Name", required = true) SchemaBranch schemaBranch) {
+        try {
+            SchemaBranch createdSchemaBranch = schemaRegistry.createSchemaBranch(schemaVersionId, schemaBranch);
+            return WSUtils.respondEntity(createdSchemaBranch, Response.Status.OK) ;
+        } catch (SchemaBranchAlreadyExistsException e) {
+            return WSUtils.respond(Response.Status.CONFLICT, CatalogResponse.ResponseMessage.ENTITY_CONFLICT,  e.getMessage());
+        } catch (SchemaNotFoundException e) {
+            return WSUtils.respond(Response.Status.BAD_REQUEST, CatalogResponse.ResponseMessage.ENTITY_NOT_FOUND,  schemaVersionId.toString());
+        } catch (Exception ex) {
+            LOG.error("Encountered error while creating a new branch with name: [{}], version : [{}]", schemaBranch.getName(), schemaVersionId, ex);
+            return WSUtils.respond(Response.Status.INTERNAL_SERVER_ERROR, CatalogResponse.ResponseMessage.EXCEPTION, ex.getMessage());
+        }
+    }
+
+    @POST
+    @Path("/schemas/{version}/merge")
+    @ApiOperation(value = "Merge a schema version to master given its version id", tags = OPERATION_GROUP_SCHEMA)
+    @UnitOfWork
+    public Response mergeSchemaVersion(@ApiParam(value = "Details about schema version",required = true) @PathParam("version") Long schemaVersionId,
+                                       @ApiParam(value = "Merge strategy for merging the schema version")
+                                       @QueryParam("strategy") @DefaultValue(DEFAULT_SCHEMA_VERSION_MERGE_STRATEGY) String strategy) {
+        try {
+            SchemaIdVersion schemaIdVersion = schemaRegistry.mergeSchemaVersion(schemaVersionId, SchemaVersionMergeStrategy.valueOf(strategy));
+            return WSUtils.respondEntity(schemaIdVersion, Response.Status.OK);
+        } catch (SchemaNotFoundException e) {
+            return WSUtils.respond(Response.Status.NOT_FOUND, CatalogResponse.ResponseMessage.ENTITY_NOT_FOUND,  schemaVersionId.toString());
+        } catch (IncompatibleSchemaException e) {
+            return WSUtils.respond(Response.Status.BAD_REQUEST, CatalogResponse.ResponseMessage.INCOMPATIBLE_SCHEMA, e.getMessage());
+        } catch (Exception ex) {
+            LOG.error("Encountered error while merging a schema version to {} branch with version : [{}]",SchemaRegistryConstants.MASTER_BRANCH, schemaVersionId, ex);
+            return WSUtils.respond(Response.Status.INTERNAL_SERVER_ERROR, CatalogResponse.ResponseMessage.EXCEPTION, ex.getMessage());
+        }
+    }
+
+    @DELETE
+    @Path("/branch/{branch_id}/delete")
```

Comment:
Path can be "schemas/branches/{branch_id}/" and no need to have delete as DELTE method already implies that.

Reply:
Done